### PR TITLE
allow teacher use CS1GD1WD1 freely in China

### PR DESCRIPTION
**Pull request by wjllance**
please give some improve advice

### CODE REVIEW

File: app/templates/courses/teacher-courses-view.jade [hunk 1]
```diff
@@ -71,7 +71,8 @@ block content
                             = " "
                             span
                               = i18n(level.attributes, 'name').replace('Course: ', '')
-              a.play-level-button.btn.btn-lg.btn-primary
+              - var disablePlayButtons = !view.paidTeacher && features.china && course.get('campaignID') != "55b29efd1cd6abe8ce07db0d" && course.get('campaignID') != "5789236960deed1f00ec2ab8" && course.get('campaignID')  != "578913f2c8871ac2326fa3e4"
+              a.play-level-button.btn.btn-lg.btn-primary(disabled=disablePlayButtons)
```
Comment (AndrewJakubowicz) [inline app/templates/courses/teacher-courses-view.jade]: Why are we making it possible to disable this button?
Do we want to block teachers from accessing the other campaigns if they haven't paid?

File: app/templates/courses/teacher-courses-view.jade [hunk 2]
```diff
@@ -101,7 +102,7 @@ mixin course-info(course)
         i(data-i18n="courses.web_dev_language_transition")
 
     if me.isTeacher() || view.ownedClassrooms.size() || me.isAdmin()
-      - var disableButtons = !course.get('free') && !view.paidTeacher;
+      - var disableButtons = !view.paidTeacher && (features.china ? (course.get('campaignID') != "55b29efd1cd6abe8ce07db0d" && course.get('campaignID') != "5789236960deed1f00ec2ab8" && course.get('campaignID')  != "578913f2c8871ac2326fa3e4") : !course.get('free'))
```
Comment (AndrewJakubowicz) [inline app/templates/courses/teacher-courses-view.jade]: See my queries above.
This check is hard to understand. One option is to move this onto the User model. Then the check could be:

`- var disableButtons = me.canAccessCampaignSolutions(campaignID)`

Comment (AndrewJakubowicz) [reply app/templates/courses/teacher-courses-view.jade]: Then we can write tests to check this behavior is correct.

Comment (wjllance) [reply app/templates/courses/teacher-courses-view.jade]: This is a Chinese demand that teachers  and students should only allow to access that 3 campaign freely, you refer to the asana task

Comment (wjllance) [reply app/templates/courses/teacher-courses-view.jade]: Can we distinguish user's nation (Features.china)inside the user model?
If not, maybe we need pass the flag features.china to the function?

**Comment by AndrewJakubowicz** [comment]
We can access features globally.

**Comment by AndrewJakubowicz** [comment]
I'm unsure this PR will address your requirements.
Teachers can already freely access all the levels and campaigns.
How are students affected by this change? I'm not sure I understand. 
What testing have you done?

**Comment by wjllance** [comment]
Sorry I didn’t speak clearly. For teachers, we want to prevent they from access other levels except the first 3, I have already test about that. For students, I haven’t make any change yet, maybe I will finish that today.

**Comment by wjllance** [comment]
> We can access features globally and you'll find examples in the user model file. (On phone so can't send link)
> […](#)

How about the paid flag, how can we know if a user (student or teacher or home user) has paid

**Comment by wjllance** [comment]
> See comments above.
> Would you be able to add tests.
> Thanks!

How to add tests-.-

**Comment by nwinter** [comment]
I don't think we need to lock things down on the server-side; we want a quick solution here, not an exhaustive one. So client-side restrictions are fine. Do we do much in the way of client-side testing for things like this?

**Comment by AndrewJakubowicz** [comment]
My main concern is that it's a big check happening in the template.
I worry about it being difficult  to maintain and in the future affecting unintended users.

I would be happier if we added this as a method to the User similar to the feature flag methods we have [here](https://github.com/codecombat/codecombat/blob/cad6aea24ba6f26414a8cee6d9f354fabbb45b81/app/models/User.coffee#L526).

Then the name of the method makes it much easier to follow what is going on.

We definitely don't do enough tests. But as we're trying to change that it shouldn't be too hard to add a test to the [user spec file](https://github.com/codecombat/codecombat/blob/1285e0a5b6b57079ad711bf095da053458d5c9b8/test/app/models/User.spec.coffee).
If it's not working then we can overlook testing. But a user method would be far more ideal than the template conditional.

File: app/models/User.coffee [hunk 3]
```diff
@@ -524,6 +524,7 @@ module.exports = class User extends CocoModel
   useGoogleAnalytics: -> not (features?.chinaInfra ? false)
   # This flag is set globally for our China server.
   showChinaVideo: -> features?.china ? false
+  canAccessCampaignFreelyFromChina: (campaignID) -> campaignID == "55b29efd1cd6abe8ce07db0d" or campaignID == "5789236960deed1f00ec2ab8" or campaignID == "578913f2c8871ac2326fa3e4"
```
Comment (wjllance) [inline app/models/User.coffee]: I think this is all I can move to user model

Comment (AndrewJakubowicz) [reply app/models/User.coffee]: This is good.

**Pull request merged**